controllers/nstemplateset/nstemplateset_controller_test.go

```diff
@@ -1766,45 +1786,78 @@ func getTemplateTiers(decoder runtime.Decoder) ([]runtime.Object, error) {
 					default:
 						tmplContent = test.CreateTemplate(test.WithObjects(ns, rb, role), test.WithParams(username))
 					}
+
 				}
-				tmplContent = strings.ReplaceAll(tmplContent, "nsType", typeName)
-				tmpl := templatev1.Template{}
-				_, _, err := decoder.Decode([]byte(tmplContent), nil, &tmpl)
+				tierTmpl, err := createTierTemplate(decoder, tmplContent, tierName, typeName, revision)
 				if err != nil {
 					return nil, err
 				}
-				tierTmpl := &toolchainv1alpha1.TierTemplate{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      strings.ToLower(fmt.Sprintf("%s-%s-%s", tierName, typeName, revision)),
-						Namespace: test.HostOperatorNs,
-					},
-					Spec: toolchainv1alpha1.TierTemplateSpec{
-						TierName: tierName,
-						Type:     typeName,
-						Revision: revision,
-						Template: tmpl,
-					},
-				}
 				tierTemplates = append(tierTemplates, tierTmpl)
 			}
 		}
 	}
+	// ---- appstudio templates ----
+	// namespace template
+	tierTmpl, err := createTierTemplate(decoder,
+		test.CreateTemplate(test.WithObjects(ns), test.WithParams(namespace, username)),
+		"appstudio", "appstudio", "abcde11")
+	if err != nil {
+		return nil, err
+	}
+	tierTemplates = append(tierTemplates, tierTmpl)
+	// space roles templates
+	tierTmpl, err = createTierTemplate(decoder,
+		test.CreateTemplate(test.WithObjects(spaceAdmin, spaceAdminRb), test.WithParams(namespace, username)),
+		"admin", "appstudio", "abcde11")
+	if err != nil {
+		return nil, err
+	}
+	tierTemplates = append(tierTemplates, tierTmpl)
+	tierTmpl, err = createTierTemplate(decoder,
+		test.CreateTemplate(test.WithObjects(spaceViewer, spaceViewerRb), test.WithParams(namespace, username)),
+		"viewer", "appstudio", "abcde11")
+	if err != nil {
+		return nil, err
+	}
+	tierTemplates = append(tierTemplates, tierTmpl)
+
 	return tierTemplates, nil
 }
 
+func createTierTemplate(decoder runtime.Decoder, tmplContent string, tierName, typeName, revision string) (*toolchainv1alpha1.TierTemplate, error) {
+	tmplContent = strings.ReplaceAll(tmplContent, "NSTYPE", typeName)
+	tmpl := templatev1.Template{}
+	_, _, err := decoder.Decode([]byte(tmplContent), nil, &tmpl)
+	if err != nil {
+		return nil, err
+	}
+	return &toolchainv1alpha1.TierTemplate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      strings.ToLower(fmt.Sprintf("%s-%s-%s", tierName, typeName, revision)),
+			Namespace: test.HostOperatorNs,
+		},
+		Spec: toolchainv1alpha1.TierTemplateSpec{
+			TierName: tierName,
+			Type:     typeName,
+			Revision: revision,
+			Template: tmpl,
+		},
+	}, nil
+}
+
 var (
 	ns test.TemplateObject = `
 - apiVersion: v1
   kind: Namespace
   metadata:
-    name: ${USERNAME}-nsType
+    name: ${USERNAME}-NSTYPE
```

Comment:
just out of curiosity - why this change?

Reply:
because I think it is more visible (ie, understandable) if the character sequence is fully capitalised. But it's just a cosmetic change.